omnibus/config/software/chefdk.rb

```diff
@@ -20,13 +20,19 @@
 # For the specific super-special version "local_source", build the source from
 # the local git checkout. This is what you'd want to occur by default if you
 # just ran omnibus build locally.
-version("local_source") { source path: "#{project.files_path}/../.." }
-#TODO: Check if I need to exclude any current omnibus generated files.
-# options: {:exclude => [some paths]}
+version("local_source") do
+  source path: "#{project.files_path}/../.."
+         # Since we are using the local repo, we try to not copy any files
+         # that are generated in the process of bundle installing omnibus.
+         # If the install steps are well-behaved, this should not matter
+         # since we only perform bundle and gem installs from the
+         # omnibus cache source directory, but we do this regardless
+         # to maintain consistency between what a local build sees and
+         # what a github based build will see.
+         options: { exclude: [ "omnibus/vendor" ] }
```

Comment:
Should we just exclude the complete `omnibus` directory? We really don't want it copied over when we are targeting the Chef application code.